backend/src/Controllers/AttachmentsController.php

```diff
@@ -31,57 +32,76 @@ class AttachmentsController extends ControllerAbstract {
 	 * @param AttachmentsRepository $attachments
 	 * @param AttachmentTransform $transformer
 	 */
-	public function __construct( AttachmentsRepository $attachments, AttachmentTransform $transformer ) {
+	public function __construct(AttachmentsRepository $attachments, AttachmentTransform $transformer)
+	{
 		$this->attachments = $attachments;
 		$this->transformer = $transformer;
 	}
 
 	/**
 	 * Register routes.
 	 */
-	public function register_routes() {
+	public function register_routes()
+	{
 		$this->route(
-			'/attachments', [
+			'/attachments',
+			[
 				[
 					'methods'             => WP_REST_Server::READABLE,
-					'callback'            => [ $this, 'index' ],
+					'callback'            => [$this, 'index'],
+					'permission_callback' => function () {
+						return current_user_can('edit_published_posts');
+					},
+				],
+
+			]
+		);
+
+		$this->route(
+			'/attachments/upload',
```

Comment:
@shahidajmeri786 do we still need this route and the related method? It looks like it's not being used.

Reply:
Removed.